conans/test/functional/cppstd/default_cppstd_test.py

```diff
@@ -0,0 +1,182 @@
+# coding=utf-8
+
+import json
+import os
+import textwrap
+import unittest
+
+from conans.client.build.cppstd_flags import cppstd_default
+from conans.client.tools import environment_append, save, load
+from conans.test.utils.deprecation import catch_deprecation_warning
+from conans.test.utils.test_files import temp_folder
+from conans.test.utils.tools import TestClient
+
+
+class DefaultCppTestCase(unittest.TestCase):
+    compiler = "gcc"
+    compiler_version = "7"
+
+    default_profile = textwrap.dedent("""
+        [settings]
+        os=Linux
+        arch=x86
+        compiler={}
+        compiler.version={}
+        compiler.libcxx=libstdc++11
+        """.format(compiler, compiler_version))
+
+    conanfile = textwrap.dedent("""
+        from conans import ConanFile
+
+        class Library(ConanFile):
+            settings = "{settings}"
+
+            def configure(self):
+                cppstd = self.settings.get_safe("cppstd")
+                compiler_cppstd = self.settings.get_safe("compiler.cppstd")
+                self.output.info(">> settings: {{}}".format(self.settings.fields))
+                self.output.info(">> cppstd: {{}}".format(cppstd))
+                self.output.info(">> compiler.cppstd: {{}}".format(compiler_cppstd))
+        """)
+
+    id_default = "d17189cfe7b11efbc5d701339a32d203745f8b81"
+
+    def run(self, *args, **kwargs):
+        default_profile_path = os.path.join(temp_folder(), "default.profile")
+        save(default_profile_path, self.default_profile)
+        with environment_append({"CONAN_DEFAULT_PROFILE_PATH": default_profile_path}):
```

Comment:
This looks complicated, why not directly writing the home/profiles/default profile?

Reply:
Why? It is a documented way to override the default profile: _"[TIP:] using the environment variable `CONAN_DEFAULT_PROFILE_PATH`"_. I think this is the easiest way to be sure that every `TestClient` created in these tests will use the same profile...